<launch>
  
  <node pkg="rosserial_python" name="Arduino" type="serial_node.py" output="screen">
    <param name="port"  type="string" value="/dev/ttyACM0"/>  
    <param name="baud" type="string" value="115200"/>
  </node>  
  
  <node pkg="test_mapping" name="IMU" type="get_imu">
    <param name="IMU_PORT"  type="string" value="/dev/ttyUSB1" />
  </node>  
<!--
  <node pkg="rplidar_ros"   name="RPLidar" type="rplidarNode"  >
    <param name="serial_port"         type="string" value="/dev/ttyUSB0"/>
    <param name="serial_baudrate"     type="int"    value="115200"/>
    <param name="frame_id"            type="string" value="scan"/>
    <param name="inverted"            type="bool"   value="false"/>
    <param name="angle_compensate"    type="bool"   value="true"/>
  </node>
-->
  <node pkg="test_mapping" name="raw_data_to_tf" type="raw_data_to_tf"  />
 
  <!-- <node pkg="gmapping" name="Gmapping" type="slam_gmapping"/> -->

  <node pkg="tf" type="static_transform_publisher" name="base_to_scan_broadcaster" args="0 0 0 3.1415 0 0 base_link scan 100" />
  


</launch>




<!--
 <param name="use_sim_time" value="true"/>
    <node pkg="gmapping" type="slam_gmapping" name="Gmapping" >
       <remap from="scan" to="base_scan"/>

      <param name="map_update_interval" value="5.0"/>
      <param name="maxUrange" value="16.0"/>

      <param name="sigma" value="0.05"/>
      <param name="kernelSize" value="1"/>
      <param name="lstep" value="0.05"/>
      <param name="astep" value="0.05"/>
      <param name="iterations" value="5"/>
      <param name="lsigma" value="0.075"/>
      <param name="ogain" value="3.0"/>
      <param name="lskip" value="0"/>
      <param name="srr" value="0.1"/>
      <param name="srt" value="0.2"/>
      <param name="str" value="0.1"/>
      <param name="stt" value="0.2"/>
      <param name="linearUpdate" value="1.0"/>
      <param name="angularUpdate" value="0.5"/>
      <param name="temporalUpdate" value="3.0"/>
      <param name="resampleThreshold" value="0.5"/>
      <param name="particles" value="30"/>
      <param name="xmin" value="-50.0"/>
      <param name="ymin" value="-50.0"/>
      <param name="xmax" value="50.0"/>
      <param name="ymax" value="50.0"/>
      <param name="delta" value="0.05"/>
      <param name="llsamplerange" value="0.01"/>
      <param name="llsamplestep" value="0.01"/>
      <param name="lasamplerange" value="0.005"/>
      <param name="lasamplestep" value="0.005"/>
    </node>
-->
 <!-- Run the map server --> 
 <!--
    <node name="map_server" pkg="map_server" type="map_server" args="$(find my_map_package)/my_map.pgm my_map_resolution"/>
 -->
 
 <!--- Run AMCL --> 
 <!--
    <include file="$(find amcl)/examples/amcl_omni.launch" />

   <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find my_robot_name_2dnav)/costmap_common_params.yaml" command="load" ns="global_costmap" /> 
    <rosparam file="$(find my_robot_name_2dnav)/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find my_robot_name_2dnav)/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find my_robot_name_2dnav)/global_costmap_params.yaml" command="load" /> 
    <rosparam file="$(find my_robot_name_2dnav)/base_local_planner_params.yaml" command="load" />
 </node>
 -->